### PR TITLE
10.9.0 Changelog

### PR DESCRIPTION
**Pull request by chrisbreiding**
- Adds 10.9.0 changelog
- Fixes some issues with the 10.8.0 changelog

### CODE REVIEW

**Review by ZachJW34 — APPROVED** [review]

File: content/_changelogs/10.9.0.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  information.
+- Added support for re-using session data cached by
+  [`cy.session()`](/api/commands/session) across specs via the
+  `cacheAcrossSpecs` option. Addresses
+  [#17710](https://github.com/cypress-io/cypress/issues/17710).
+- Added support for advanced dev server configuration via an async function that
+  can optionally modify the dev server config. Addresses
+  [#23302](https://github.com/cypress-io/cypress/issues/23302).
+
+**Bugfixes:**
```
Comment (mike-plummer) [inline content/_changelogs/10.9.0.md:20]: Missing bugfix [#23409](https://github.com/cypress-io/cypress/issues/23409), and we're expecting to get feature [#21720](https://github.com/cypress-io/cypress/issues/21720) in today

Comment (chrisbreiding) [reply content/_changelogs/10.9.0.md:26]: Added #23409 in https://github.com/cypress-io/cypress-documentation/pull/4752/commits/815d89147c62bcb37751467060979be289701e15. I'll make sure #21720 is added once it's merged.

**Comment by emilyrohrbough** [comment]
Need entry for https://github.com/cypress-io/cypress/pull/23624

**Comment by chrisbreiding** [comment]
> Need entry for https://github.com/cypress-io/cypress/pull/23624

Good catch! Addressed in https://github.com/cypress-io/cypress-documentation/pull/4752/commits/93784c86ea181ed21578926233ebd5fa0cb88026

File: content/_changelogs/10.9.0.md (hunk 1 above)
Comment (chrisbreiding) [reply content/_changelogs/10.9.0.md:37]: Added #21720: https://github.com/cypress-io/cypress-documentation/pull/4752/commits/9c9f4154e4d078e41217451a906a0de7f1064314

**Pull request merged**